### DISCUSSION TITLE
State of React Native 2024 Preview / Call for Questions

### DISCUSSION THREAD

**Discussion by bbarthec** [software-mansion/state-of-react-native#9] [Preview / Call for Questions]
## [Preview of the 2024 survey](https://survey.stateofreactnative.com/)

At Software Mansion, we're making progress on the next edition of the [State of React Native survey](https://stateofreactnative.com/) and present you **the preview** of the upcoming 2024 edition. As we did last year, we invite all of you to join us in making the the question set the best it can be. 

Aside from the question set changes, the other most notable change this year is the framework upgrade and visual redesign. [@SachaG's suggestion from last year](https://github.com/software-mansion/state-of-react-native/discussions/6#discussioncomment-7521713) made us realise that we've been missing so many good parts of the original surveys. We're building the `State of React Native survey` based on the work of the @Devographics team, and we're more than happy to take advantage of all the features they've coded into the survey framework. This year we've made it possible to use `reading list`, `question skipping`, a more detailed response system, etc., effectively aligning the `State of React Native` survey with its original counterparts. 

## Call for Questions

At Software Mansion, we're iterating on the question set to get the best coverage of the most important React Native-oriented aspects. At this point, we're still discussing the layout of the sections as well as the individual questions we have.
We also value the feedback we can get from the React Native community, so we invite you all to suggest changes to the content of the survey to make it even better. Don't forget to justify your suggestions!

We'll do our best to review the suggestions on a weekly basis and respond accordingly by either accepting or rejecting them.

## Expected timeline

The preview period will run through the last week of November. The 2024 survey will go live the first week of December and would run until January 6, 2025.

The results will be published sometime in the second half of January 2025 or early February 2025.

### Meaningful content changes we've introduced this year

The list of changes is quite lengthy and it's best to just directly go to the preview and see inspect the questions live.

**Comment by slorber** [comment]
Hey @bbarthec 

Here's my review of the survey. Overall it looks great but there are probably a few little things to tweak.

I added some possible commenters for each section. 
Not always easy to pick names but this can serve as an inspiration.
I can also comment if you want, although I don't really have any recent RN prod app experience.

---



# State of RN 2024 - Survey Review

## Intro

Maybe find a better non-X link for "continues to grow"?
Original: https://x.com/Baconbrix/status/1703816157831045411
Possible alt, but only Expo: https://evanbacon.dev/blog/expo-2024


Update Survey design GH discussion link to this 2024 issue instead of 2023


## Background

Some questions are not easy to answer for me, as someone who's not actively working on any RN app in 2024.
For example:
- choosing my team size
- mobile app number of users
- mobile OS version target
I think I'm not alone. There are also contractors working for many apps at the same time.
Maybe give an escape hatch answer choice "N/A", "Other" or something to avoid biasing other answers. 
There's the ability to not answer a question, but I think people will select a wrong answer by default.
Edit: I only noticed the "skip question" later 😅


Destkop OS => Other choice, free field?


Platforms => Skia (Linux & macOS) feels like a weird choice that people will pick despite not targeting desktop. Maybe juste put "Linux" ?



## Platform APIs


N/A

Commenters:
- https://x.com/mozzius
- https://x.com/zoontek
- https://x.com/hirbod_dev
- https://x.com/Baconbrix


## State Management

Maybe remove Apollo / React Query since they are already in data fetching


Commenters:
- https://x.com/jmeistrich
- https://x.com/jamonholmgren
- https://x.com/dai_shi


## Data Fetching

It's weird to have the same libs in both state management and data fetching

The alternative is to split this in 2:
- Async State Management: Apollo, Relay, SWR, React Query, RTK Query
- Data Fetching: fetch, axios, tRPC


Commenters:
- https://x.com/alexdotjs
- https://x.com/TkDodo




## Navigation


Commenters:
- https://x.com/Baconbrix
- https://x.com/satya164
- https://x.com/FernandoTheRojo




## Styling

Remove "tailwind-rn" (too niche / unmaintained)

Add stylex ? (afaik at least Meta uses it in a cross-platform way with React Strict DOM)
Add React Strict DOM?



Commenters:
- https://x.com/naman34
- https://github.com/necolas
- https://x.com/DanStepanov
- https://x.com/natebirdman
- https://x.com/jpudysz




## Component Libraries


Commenters:
- https://x.com/natebirdman
- https://x.com/sanketsahu
- https://x.com/_happiryu






## Animations


Rename section to something that encapsulate graphics too?

React Three Fiber and Skia are not really related to animations


Add React Native Gesture Handler (+ RN Screens?)

Add React Native WGPU: https://github.com/wcandillon/react-native-webgpu

Add React Native Godot (note there are 2 Godot integrations WIP, and only one is open sourced afaik https://github.com/calico-games/react-native-godot)



Commenters:
- https://x.com/kzzzf
- https://x.com/wcandillon
- https://x.com/reactiive_
- https://x.com/mironcatalin
- https://x.com/aarongrider





## Debugging

Might be hard to figure out which debugger you actually use 😅

Commenters:
- https://x.com/alxhnt
- https://x.com/cedric_dev
- https://x.com/sebmarkbage




## App Aspecs


Analytics => posthog, btw they recently launched session replay RN SDK
https://posthog.com/blog/mobile-session-replay
https://github.com/PostHog/posthog-react-native-session-replay

Data visualization: 
- skia
- https://github.com/margelo/react-native-graph


Commenters:
- https://x.com/mrousavy
- https://x.com/notbrent
- https://x.com/thymikee
- https://x.com/mdj_dev





## RN Features

Add: Concurrent React (transitions, useDeferredValue)


Commenters:
- https://x.com/CipolleschiR
- https://x.com/cortinico
- https://x.com/tsapeta
- https://x.com/birch_js


## Deployment


Add Expo Workflows?


Commenters:
- https://x.com/jonsamp
- https://x.com/jamonholmgren
- https://x.com/flexbox_



## RN Tools


Yarn => Yarn v1 ?


Commenters:
- https://x.com/ccheever
- https://x.com/schlimmson
- https://x.com/VadimNotJustDev




## Alternatives

Add Tauri / Electron?


Commenters:
- https://x.com/schlimmson
- https://x.com/mo__javad
- https://x.com/birch_js
- https://x.com/hugemathguy



## Resources



Expo "Exposition" blog is the old medium-based one
The new one is not called "Exposition": https://expo.dev/blog

Kent C. Dodds is not realy doing RN. It's a bit weird if it's the only one teaching React core here. There are many others.

Other RN-related newsletters: notJust.dev, Galaxies.dev


Other courses: Galaxies, React Native Mastery
Not sure Start React Native (William) is super active and up to date nowadays


Commenters:
- https://x.com/jherr
- https://x.com/reactiive_
- https://x.com/wcandillon


## Opinions


What do you feel currently is missing from React Native?
"Android shadows" is in RN 0.76 no?

Advantages: "learn once, built everywhere"


Commenters:
- https://x.com/theo
- https://x.com/rickhanlonii
- https://x.com/kzzzf
- https://x.com/grabbou

**Reply by bbarthec** [reply]
@slorber, thank you for pointing out all of these aspects 👏 I'll take a deep look into the suggestions and incorporate them accordingly into the survey 💪

**Comment by huntie** [comment]
Hey @bbarthec, thanks again for preparing the 2024 survey and taking our input!

Coming in with some suggestions for the **Debugging and profiling** page.

- Let's subdivide the list of tools — I think it's worth grouping and reordering, to help signal what things are standalone tools and what are built-in/DevTools sub-features. In particular, React DevTools and React Native DevTools are ambiguously placed.
- Right now, this page is a long set of question boxes with the sole result of "which tool do you use". Let's add a second section with the goal of discovering what users want to achieve when they debug.

By the way, the survey UX is top notch this year! 👏🏻 I **love** the positive/negative annotations!

cc @hoxyq @robhogan

---

## Debugging and profiling

> 💡 **Changes**: Group tools items under two subheadings. Add "Debugging intent" section. My annotations are in **[square brackets]**.

### [Add section] Tools and features

- Debugging clients and tools
    - React Native DevTools
    - Reactotron
    - Radon IDE
    - Legacy JavaScript runtime debuggers
    - Flipper
    - Native debugging
    - Native profiling
    - **[Add]** Expo DevTools Plugins
    - Flashlight
- Debugging features
    - Console API
    - React DevTools
    - Element Inspector
    - Performance Monitor
    - **[Rename]** Performance API (react-native-performance)
    - **[Add]** Network Inspector (Expo)
    - **[Delete LogBox]**
- Other solutions [freeform]

### [Add section] Debugging intent

**[Prompt]** *What do you want to do when debugging your app?*

[Sub bullet points are descriptions]

- View app logs
    - "Viewing and inspecting app logs."
- Inspect layout and styles
  - "Inspecting the element tree, highlighting elements in the UI, changing styles in realtime."
- Inspect code behaviour
  - "Step-through breakpoint debugging, evaluating expressions in the JavaScript console."
- Understand React behaviour
  - "Inspecting React state, analyzing performance issues - e.g. re-renders."
- Inspect network requests
- Profile your app
  - "Monitoring resource usage (Memory, CPU), analyzing execution time of functions."
- Understand the behaviour of third party JS libraries
  - "Use specific dev plugins for third party libraries or use cases (e.g. Redux DevTools). Available today via Expo DevTools Plugins."
- Other [freeform]

(**Edit**: I originally forgot about the "pain points" question — let's keep that too! 🙂)
